src/common/walFilter/greenplumCommon.h

```diff
@@ -0,0 +1,105 @@
+#ifndef PGBACKREST_GREENPLUMCOMMON_H
+#define PGBACKREST_GREENPLUMCOMMON_H
+
+#include <stdint.h>
+#include "postgres/interface/static.vendor.h"
+
+// Common macros and constants for different versions of Greenplum
+
+// types
+typedef uint32_t TimeLineID;
+typedef uint64_t XLogRecPtr;
+typedef uint8_t RmgrId;
+typedef uint32_t pg_crc32;
+typedef uint32_t TransactionId;
+typedef uint32_t BlockNumber;
+
+// constants
+#define XLOG_BLCKSZ 32768
+/* This flag indicates a "long" page header */
+#define XLP_LONG_HEADER             0x0002
+#define MAXIMUM_ALIGNOF 8
+#define BLCKSZ 32768
+#define XLR_MAX_BKP_BLOCKS      4
+#define XLP_FIRST_IS_CONTRECORD     0x0001
+#define XLP_FIRST_IS_OVERWRITE_CONTRECORD 0x0008
+
+#define RM_XLOG_ID                      0
+#define XLOG_SWITCH                     0x40
+#define XLOG_NOOP                       0x20
+#define MAXPGPATH       1024
+// macros
+#define MAXALIGN(LEN)           TYPEALIGN(MAXIMUM_ALIGNOF, (LEN))
+#define TYPEALIGN(ALIGNVAL,LEN)  \
+    (((uintptr_t) (LEN) + ((ALIGNVAL) - 1)) & ~((uintptr_t) ((ALIGNVAL) - 1)))
+
+#define SizeOfXLogShortPHD  MAXALIGN(sizeof(XLogPageHeaderData))
+#define SizeOfXLogLongPHD   MAXALIGN(sizeof(XLogLongPageHeaderData))
+#define XLogPageHeaderSize(hdr)     \
+    (((hdr)->xlp_info & XLP_LONG_HEADER) ? SizeOfXLogLongPHD : SizeOfXLogShortPHD)
+#define SizeOfXLogRecord    MAXALIGN(sizeof(XLogRecord))
+#define XLogRecGetData(record)  ((char*) (record) + SizeOfXLogRecord)
+#define XLR_BKP_BLOCK(iblk)     (0x08 >> (iblk))        /* iblk in 0..3 */
+#define Min(x, y)       ((x) < (y) ? (x) : (y))
+
+#define XLOG_SEG_SIZE (64 * 1024 * 1024)
```

Comment:
Greenplum [may be configured](https://github.com/arenadata/gpdb/blob/2ce272a173b3353bcb45723ed8b9aa887025f18d/configure#L3858-L3894) with different sizes